tests/core/test_lightning_module.py

```diff
@@ -358,3 +360,26 @@ def configure_optimizers(self):
     )
 
     trainer.fit(model)
+
+
+@RunIf(min_gpus=1)
+def test_device_placement(tmpdir):
+
+    model = BoringModel()
+    trainer = Trainer(default_root_dir=tmpdir, fast_dev_run=True, gpus=1)
+    trainer.fit(model)
+
+    def assert_device(device: torch.device) -> None:
+        assert model.device == device
+        for p in model.parameters():
+            assert p.device == device
+
+    assert_device(torch.device("cpu"))
+    model.to(torch.device("cuda:0"))
+    assert_device(torch.device("cuda:0"))
+
+    trainer = Trainer(default_root_dir=tmpdir, fast_dev_run=True, gpus=1)
```

Comment:
Do we need to create a new trainer instance?

```suggestion
    trainer = Trainer(default_root_dir=tmpdir, fast_dev_run=True, gpus=1)
```